src/pages/Posts.tsx

```diff
@@ -0,0 +1,112 @@
+// import useApiHook from "../utils/useApiHook";
+import { Post, PostWithComment, ViewPost } from "../interfaces";
+import { UserContentContext } from "../context/UserContentContext";
+import SinglePost from "./SinglePost";
+
+function Posts() {
+    const userId: number = useLocationId(); // Get the id parameter for URL
+    const [isLoading, setIsLoading] = useState(true); // If the posts are loading or not
+    const [viewablePost, setViewablePost] = useState<ViewPost>({
+        view: false,
+        post: {
+            post: {
+                userid: 0,
+                title: "",
+                body: "",
+                id: 0,
+            },
+            comments: [],
+        },
+        index: -1,
+    });
+
+    const { state, dispatch } = useContext(UserContentContext);
+    const { user_posts } = state;
+    useEffect(() => {
+        // If the posts in the state is of different user, then update the state,
+        // else update render with the same state
+        if (userId !== user_posts.userId) {
+            axios
+                .get(`/users/${userId}/posts`)
+                .then((response) => {
+                    // Create an array where each object has a post and an empty comment array
+                    let posts_without_comments = response.data;
+                    let posts_with_comments: Array<PostWithComment> = [];
+                    posts_without_comments.forEach((val: Post) => {
+                        posts_with_comments.push({
+                            post: val,
+                            comments: null,
+                        });
+                    });
+                    dispatch({
+                        type: "SET_POSTS",
+                        payload: { userId: userId, posts: posts_with_comments },
+                    });
+                    setIsLoading(false);
+                })
+                .catch((err) => {});
+        } else {
+            setIsLoading(false);
+        }
+    }, [userId, user_posts, dispatch]);
+
+    return (
+        <div>
+            <div style={{ marginTop: "20px" }}>
+                <Link to={`/users/${userId}/todos`}>
+                    <button>Go to Todos</button>
+                </Link>
+                <Link to={`/users/${userId}/new_post?edit=false`}>
+                    <button>Create a post</button>
+                </Link>
+            </div>
+            {/* {error && <div>{error.message}</div>} */}
+            {viewablePost.view && (
+                <SinglePost
+                    post={viewablePost.post}
+                    setter={setViewablePost}
+                    index={viewablePost.index}
+                />
+            )}
+            {isLoading ? (
+                <div>Loading...</div>
+            ) : (
+                <>
+                    <h1 style={{ textAlign: "center" }}>Posts</h1>
+                    {user_posts.posts &&
+                        user_posts.posts.map((val: PostWithComment, idx) => (
```

Comment:
Create a separarte function called `renderUserPosts = useCallback(() => user_posts.map ,[])` to handle rendering here